prow/plugins/approve/testfreeze/testfreeze.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testfreeze
+
+import (
+	"strings"
+
+	"github.com/blang/semver"
+	git "github.com/go-git/go-git/v5"
+	gitconfig "github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	gitmemory "github.com/go-git/go-git/v5/storage/memory"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Checker is the main structure of checking if we're in Test Freeze.
+type Checker struct {
+	impl impl
+	log  *logrus.Entry
+}
+
+// Result is the result returned by `InTestFreeze`.
+type Result struct {
+	// InTestFreeze is true if we're in Test Freeze.
+	InTestFreeze bool
+
+	// Branch is the found latest release branch.
+	Branch string
+
+	// Tag is the latest minor release tag to be expected.
+	Tag string
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+//counterfeiter:generate . impl
+type impl interface {
+	ListRefs(*git.Remote) ([]*plumbing.Reference, error)
+}
+
+type defaultImpl struct{}
+
+// NewChecker creates a new Checker instance.
+func NewChecker(log *logrus.Entry) *Checker {
+	return &Checker{
+		impl: &defaultImpl{},
+		log:  log,
+	}
+}
+
+// InTestFreeze returns if we're in Test Freeze:
+// https://github.com/kubernetes/sig-release/blob/2d8a1cc/releases/release_phases.md#test-freeze
+// It errors in case of any issue.
+func (c *Checker) InTestFreeze() (*Result, error) {
+	remote := git.NewRemote(gitmemory.NewStorage(), &gitconfig.RemoteConfig{
+		Name: "origin",
+		URLs: []string{"https://github.com/kubernetes/kubernetes"},
+	})
+
+	refs, err := c.impl.ListRefs(remote)
+	if err != nil {
+		c.log.Errorf("Unable to list git remote: %v", err)
+		return nil, errors.Wrap(err, "list git remote")
+	}
+
+	const releaseBranchPrefix = "release-"
+	var (
+		latestSemver semver.Version
+		latestBranch string
+	)
+
+	for _, ref := range refs {
+		if ref.Name().IsBranch() {
+			branch := ref.Name().Short()
+
+			// Filter for release branches
+			if strings.HasPrefix(branch, releaseBranchPrefix) {
```

Comment:
The cycle block could be more readable with guard clauses instead of the nested ifs, wdyt?

Reply:
Agreed, fixed by checking for `!strings.HasPrefix(branch, releaseBranchPrefix)` and continuing if so.